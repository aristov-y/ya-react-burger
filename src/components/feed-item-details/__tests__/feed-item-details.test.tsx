import React from 'react';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import FeedItemDetails from '../index';
import { useSelector } from 'react-redux';

jest.mock('../../feed-item-ingredients', () => () => 'FeedItemIngredients')

const store = {
  ingredients: {
    ingredients: [{
      _id: 'main01',
      price: 10
    }, {
      _id: 'main02',
      price: 12
    }]
  }
}

beforeEach(() => {
  (useSelector as jest.Mock).mockImplementation(fn => fn(store));
});

describe('FeedItemDetails', function () {
  it('should render', function () {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<FeedItemDetails
        name={"Name 1"}
        status={"pending"}
        createdAt={"2020-01-01T00:00:00"}
        ingredients={["main01"]}
      />);
    });

    expect(component.toJSON()).toMatchSnapshot();

    act(() => {
      component.update(<FeedItemDetails
        name={"Name 2"}
        status={"done"}
        createdAt={"2020-01-02T00:00:00"}
        ingredients={["main01", "main02"]}
      />)
    })

    expect(component.toJSON()).toMatchSnapshot();
  });
});
