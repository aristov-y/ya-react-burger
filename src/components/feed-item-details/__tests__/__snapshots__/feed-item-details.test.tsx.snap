// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedItemDetails should render 1`] = `
<div
  className="FeedItemDetails text text_type_main-default"
>
  <span
    className="mb-4 text_type_main-medium"
  >
    Name 1
  </span>
  <span
    className="mb-4 "
  >
    В работе
  </span>
  <div
    className="Scroller mb-4"
  >
    FeedItemIngredients
  </div>
  <div
    className="FeedItemDetailsFooter"
  >
    <span
      className="text_color_inactive"
    >
      01.01.2020, 00:00:00
    </span>
    <span
      className="text_type_digits-default"
    >
      10
      CurrencyIcon
    </span>
  </div>
</div>
`;

exports[`FeedItemDetails should render 2`] = `
<div
  className="FeedItemDetails text text_type_main-default"
>
  <span
    className="mb-4 text_type_main-medium"
  >
    Name 2
  </span>
  <span
    className="mb-4 Completed"
  >
    Выполнен
  </span>
  <div
    className="Scroller mb-4"
  >
    FeedItemIngredients
  </div>
  <div
    className="FeedItemDetailsFooter"
  >
    <span
      className="text_color_inactive"
    >
      02.01.2020, 00:00:00
    </span>
    <span
      className="text_type_digits-default"
    >
      22
      CurrencyIcon
    </span>
  </div>
</div>
`;
